x/concentrated-liquidity/genesis.go

```diff
@@ -64,11 +81,34 @@ func (k Keeper) ExportGenesis(ctx sdk.Context) *genesis.GenesisState {
 		if err != nil {
 			panic(err)
 		}
+		accumObject, err := k.getFeeAccumulator(ctx, poolI.GetId())
+		if err != nil {
+			panic(err)
+		}
+
+		totalShares, err := accumObject.GetTotalShares()
+		if err != nil {
+			panic(err)
+		}
 
-		poolData = append(poolData, &genesis.PoolData{
-			Pool:  &anyCopy,
-			Ticks: ticks,
+		genAccumObject := genesis.AccumObject{
+			Name:        getFeeAccumulatorName(poolI.GetId()),
+			Value:       accumObject.GetValue(),
+			TotalShares: totalShares,
+		}
+
+		poolData = append(poolData, genesis.PoolData{
+			Pool:        &anyCopy,
+			Ticks:       ticks,
+			AccumObject: genAccumObject,
 		})
+
+		poolId := poolI.GetId()
+		incentiveRecordsForPool, err := k.GetAllIncentiveRecordsForPool(ctx, poolId)
+		if err != nil {
+			panic(err)
+		}
+		incentiveRecords = append(incentiveRecords, incentiveRecordsForPool...)
```

Comment:
Why don't we store incentive records directly in `poolData` if we get them by pool id?

The reason we have positions separately is because we had to no way to get them from store by pool id. That can be changed now with the position key refactor

Reply:
makes sense, moved incentive records field into `PoolData`